_meta/fields.common.yml

```diff
@@ -316,3 +316,256 @@
+
+    - name: http
+      type: group
+      dynamic: false
+      fields:
+        - name: method
+          type: alias
+          path: context.request.method
+
+        - name: version
+          type: alias
+          path: context.request.http_version
+
+    # context.tags copied here (can't alias objects)
+    - name: labels
+      type: object
+      object_type: keyword
+      dynamic: true
+      description: >
+        A flat mapping of user-defined labels with string values.
+
+    - name: process
+      type: group
+      dynamic: false
+      fields:
+        - name: pid
+          type: alias
+          path: context.process.pid
+
+        - name: ppid
+          type: alias
+          path: context.process.ppid
+
+        - name: title
+          type: alias
+          path: context.process.title
+
+    - name: service
+      type: group
+      dynamic: false
+      fields:
+        # not in ECS
+        - name: environment
+          type: alias
+          path: context.service.environment
+
+        # not in ECS
+        - name: framework
+          type: group
+          fields:
+            - name: name
+              type: alias
+              path: context.service.framework.name
+
+            - name: version
+              type: alias
+              path: context.service.framework.version
+
+        # not in ECS
+        - name: language
+          type: group
+          fields:
+            - name: name
+              type: alias
+              path: context.service.language.name
+
+            - name: version
+              type: alias
+              path: context.service.language.version
+
+        - name: name
+          type: alias
+          path: context.service.name
+
+        # not in ECS
+        - name: runtime
+          type: group
+          fields:
+            - name: name
+              type: alias
+              path: context.service.runtime.name
+
+            - name: version
+              type: alias
+              path: context.service.runtime.version
+
+        - name: version
+          type: alias
+          path: context.service.version
+
+    - name: url
+      type: group
+      dynamic: false
+      fields:
+        - name: domain
+          type: alias
+          path: context.request.url.hostname
+
+        - name: fragment
+          type: alias
+          path: context.request.url.hash
+
+        - name: full
+          type: alias
+          path: context.request.url.full
+
+        - name: original
+          type: alias
+          path: context.request.url.raw
+
+        - name: path
+          type: alias
+          path: context.request.url.pathname
+
+        # context.request.url.port keyword -> long
+        - name: port
+          type: long
+          description: >
+            The port of the request, e.g. 443.
+
+        - name: query
+          type: alias
+          path: context.request.url.search
+          # TODO: multifield original.text
```

Comment:
I don't understand this comment, as I don't see any `text` in our current `url` mappings nor in the ECS schema.

Reply:
Context: in ECS, almost all fields are indexed as `keyword`. This is what most monitoring use cases need.

There's two future situations we consider:

- If we decide to add full text search support for some fields in ECS
- If a user decides they want full text search on ECS fields that aren't currently defined that way

In both cases, a `.text` multi-field should be added to the field in question. This is not a breaking change in the context of ECS, and this is not a conflict with ECS, if a user does this.

Currently however, ECS does not define any multi-field this way. If `url.original` is a field APM expects to be able to do full text search on (and for any other ECS field where this applies), you're welcome to define this multi-field.

Note that since the overwhelming majority of monitoring use cases use mostly keyword for aggregation on short values, we've flipped the usual convention around. Instead of the ElasticSearch default of doing `[field]` == `text` and `[field].keyword` == `keyword`, in ECS we do `[field]` == `keyword` and `[field].text` == `text`.

Reply:
We'd love do know which fields you're defining as `text` in APM. I do think it would be helpful to define a few as such in ECS.

Your input will help us plan this.

Reply:
This comment is actually left over from before https://github.com/elastic/ecs/pull/137, which changed a number of fields from `text` to `keyword` in ECS.  I'll remove the comment and audit the remaining fields.